test/utils.spec.js

```diff
@@ -9,3 +11,35 @@ describe('twitter', () => {
     expect(tweetURL).to.be.equal('https://archive.org/')
   })
 })
+
+describe('isValidUrl', ()=>{
+  it('should reject non-strings', ()=>{
+    let result = isValidUrl(5) || isValidUrl({}) || isValidUrl(true)
+    expect(result).to.be.false
+  })
+  it('should reject impropper protocols', ()=>{
+    let result = isValidUrl('chrome://extension')
+    expect(result).to.be.false
+
+  })
+  it('should accept propper urls', ()=>{
+    let result = isValidUrl('https://example.com/')
+    expect(result).to.be.true
+  })
+})
+
+describe('isNotExcludedUrl', () =>{
+  var excluded_urls = [
+    "localhost",
+    "0.0.0.0",
+    "127.0.0.1"
+  ];
+  it('should reject urls on the excluded list', ()=>{
+    let result = isNotExcludedUrl('https://0.0.0.0', excluded_urls)
+    expect(result).to.be.false
+  })
+  it('should accept urls not on the list', ()=>{
+    let result = isNotExcludedUrl('https://example.com', excluded_urls)
+    expect(result).to.be.true
+  })
+})
```

Comment:
a) There is room for a lot of improvement in the testing code.
b) You need to add many more test cases.
Imagine that you have a list like:
```
[
    {"url": "https;//0.0.0.0", "result": False},
    {"url:" "http://example.com", "result": True},
    ....
]
```
and then you have a for loop where you check each of these items.
The code should be MUCH more compact and it would be easier for you to add a lot of test cases.

I'm posting an example from another project to get the idea:
```
let items = [
    {
      url: 'domain:example.com',
      query: 'domain:example.com',
      queryValue: 'example.com',
      queryType: 'domain'
    },
    {
      url: 'domain:http://example.com/',
      query: 'domain:example.com',
      queryValue: 'example.com',
      queryType: 'domain'
    },
    {
      url: 'domain:http://example.com/other?a=1',
      query: 'domain:example.com',
      queryValue: 'example.com',
      queryType: 'domain'
    },
    {
      url: 'domain:example.com/other?a=1',
      query: 'domain:example.com',
      queryValue: 'example.com',
      queryType: 'domain'
    },
    {
      url: 'host:ftp://example.com/other?a=1',
      query: 'host:example.com',
      queryValue: 'example.com',
      queryType: 'host'
    },
    {
      url: 'host:example.com/other?a=1',
      query: 'host:example.com',
      queryValue: 'example.com',
      queryType: 'host'
    },
    {
      url: 'example.com/abc',
      query: 'example.com',
      queryValue: 'example.com',
      queryType: 'host'
    },
    {
      url: 'tld:com',
      query: 'tld:com',
      queryValue: 'com',
      queryType: 'tld'
    },
    {
      url: 'org',
      query: 'org',
      queryValue: 'org',
      queryType: 'host'
    }
  ];

  items.forEach(({ url, query, queryType, queryValue }) =>
    it(`should convert url: "${url}" to query: "${query}", value "${queryValue}" and type "${queryType}"`, () => {
      expect(detectSummaryQuery(url)).to.deep.equal({ query, queryType, queryValue });
    })
  );
```